src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -968,6 +978,118 @@ class spell_dru_t4_2p_bonus : public SpellScriptLoader
+                return SPELL_FAILED_NOT_SHAPESHIFT;
+            if (!player->IsInWater() && !player->GetMap()->IsOutdoors(player->GetPositionX(), player->GetPositionY(), player->GetPositionZ()))
+                return SPELL_FAILED_ONLY_OUTDOORS;
+
+            return SPELL_CAST_OK;
+        }
+
+        void Register() override
+        {
+            OnCheckCast += SpellCheckCastFn(spell_dru_travel_form_SpellScript::CheckCast);
+        }
+    };
+
+    class spell_dru_travel_form_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_dru_travel_form_AuraScript);
+
+        void AfterApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Player* player = GetTarget()->ToPlayer();
+            if (player->IsInWater())
+                player->CastSpell(player, DRUID_AQUATIC_FORM, true);
+            else if (player->IsInCombat())
+                player->CastSpell(player, DRUID_STAG_FORM, true);
+            else if ((player->GetMapId() == 530) || (player->GetMapId() == 571 && player->HasSpell(54197)) || ((player->GetMapId() == 0 || player->GetMapId() == 1) && player->HasSpell(90267)))
+            {
+                if (player->getLevel() >= 71)
+                    player->CastSpell(player, DRUID_SWIFT_FLIGHT_FORM, true);
+                else if (player->getLevel() >= 58)
+                    player->CastSpell(player, DRUID_FLIGHT_FORM, true);
+                else
+                    player->CastSpell(player, DRUID_STAG_FORM, true);
+            }
+            else
+                player->CastSpell(player, DRUID_STAG_FORM, true);
+        }
+
+        void AfterRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            Player* player = GetTarget()->ToPlayer();
+            if (GetTargetApplication()->GetRemoveMode() == AURA_REMOVE_BY_DEFAULT) //Automatic Form Swapping
+            {
+                if (player->IsInWater())
+                    player->CastSpell(player, DRUID_AQUATIC_FORM, true);
+                else if (!player->GetMap()->IsOutdoors(player->GetPositionX(), player->GetPositionY(), player->GetPositionZ()))
+                {
+                    player->RemoveAura(SPELL_DRUID_TRAVEL_FORM, ObjectGuid::Empty, 0U, AURA_REMOVE_BY_CANCEL);
+                    player->RemoveAura(DRUID_STAG_FORM, ObjectGuid::Empty, 0U, AURA_REMOVE_BY_CANCEL);
+                    player->RemoveAura(DRUID_AQUATIC_FORM, ObjectGuid::Empty, 0U, AURA_REMOVE_BY_CANCEL);
+                    player->RemoveAura(DRUID_FLIGHT_FORM, ObjectGuid::Empty, 0U, AURA_REMOVE_BY_CANCEL);
+                    player->RemoveAura(DRUID_SWIFT_FLIGHT_FORM, ObjectGuid::Empty, 0U, AURA_REMOVE_BY_CANCEL);
+                }
+                else if ((player->GetMapId() == 530) || (player->GetMapId() == 571 && player->HasSpell(54197)) || ((player->GetMapId() == 0 || player->GetMapId() == 1) && player->HasSpell(90267)))
+                {
+                    if (player->getLevel() >= 71)
```

Comment:
Is there no other requirements for Swift Flight Form anymore?
http://wowwiki.wikia.com/wiki/Swift_Flight_Form_quest_chain